### ISSUE TITLE
Improved bindings interface and API

### ISSUE THREAD

**Issue by aidanholm** [luakit/luakit#384]
- #375: API to remove binding
 - See also https://github.com/aidanholm/luakit/issues/119
 - [x] Error checking for `lousy.binds.parse_mods`
 - [x] Move bind adding/removing to `lousy.bind`
 - [x] If shadowing a pre-existing binding, replace the old binding.
   - Will currently only work if the bind string is an exact match: won't work with buffer patterns.
 - [x] Shorthand forms for Control, Shift, eg: `<C-S-x>` is the same as `<Control-Shift-x>`
 - [x] Switch bind triggers to be strings, not constructed tables
 - [x] Separate bind trigger and bind action
    - Will still keep the old method of just defining a function, for quick-and-dirty bindings.
    - Actions will have a `validate_binding` hook.
    - Have not yet converted bindings to use action tables. That would be better done with completion formats, etc:
```lua
-- In go_next_prev.lua
_M.actions.go_next = {
    func = function (w, opts) ... end,
    desc = "Go to next page",
}
--- In bookmarks.lua
_M.actions.bookmark_page = {
    func = function (w, opts) ... end,
    desc = "Bookmark a URI, default is current page",
    format = "<uri>" -- format string used by completion module to autocomplete/validate
}
```

**Comment by aidanholm** [comment]
### New action function signature:
1. `w`: the window table
2. `invoc`: invocation parameters of the action. Will include:
 - command argument string if activated via command
 - buffer text, if activated via buffer
 - the bind that triggered the action
3. `opts`: bind-time parameters

### Example use: `quickmark-go`

 - `quickmark-go` can then compare the invocation-time buffer text and the buffer pattern, and extract the quickmark ID. Effectively, you can remap this command to whatever buffer pattern you like, and it'll work seamlessly.
 - It'll also compare its command argument, if command activated: `:quickmark-go x` works.
 - If you try to bind it to something that won't work, e.g. a mouse button, its `validate_binding` will throw an error immediately.

**Comment by aidanholm** [comment]
WIP on `binds-api` branch.

**Comment by aidanholm** [comment]
Merged at c15796e2
